searx/engines/pdbe.py

```diff
@@ -0,0 +1,118 @@
+"""
+ PDBe (Protein Data Bank in Europe)
+
+ @website       https://www.ebi.ac.uk/pdbe
+ @provide-api   yes (https://www.ebi.ac.uk/pdbe/api/doc/search.html),
+                unlimited
+ @using-api     yes
+ @results       python dictionary (from json)
+ @stable        yes
+ @parse         url, title, content, img_src
+"""
+
+from json import loads
+from flask_babel import gettext
+from searx.utils import searx_useragent
+
+categories = ['science']
+
+hide_obsolete = False
+hide_unpublished = True
+
+# status codes of unpublished entries
+pdb_unpublished_codes = ['HPUB', 'HOLD', 'PROC', 'WAIT', 'AUTH', 'AUCO', 'REPL', 'POLC', 'REFI', 'TRSF', 'WDRN']
+# url for api query
+pdbe_solr_url = 'https://www.ebi.ac.uk/pdbe/search/pdb/select?'
+# base url for results
+pdbe_entry_url = 'https://www.ebi.ac.uk/pdbe/entry/pdb/{pdb_id}'
+# link to preview image of structure
+pdbe_preview_url = 'https://www.ebi.ac.uk/pdbe/static/entry/{pdb_id}_deposited_chain_front_image-200x200.png'
+
+
+def request(query, params):
+
+    params['url'] = pdbe_solr_url
+    params['method'] = 'POST'
+    # using searx User-Agent
+    params['headers']['User-Agent'] = searx_useragent()
```

Comment:
Why is `searx_useragent` is called here? If you set it, you leak that information that a searx instance made the request. If the point was to hide the user agent of the user, it is unnecessary, because searx generates a random user agent when a request is sent.

This function was introduced to authenticate for BASE engine. Doest it use the same auth method as BASE does?